internal/xpkg/reader.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2022 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package xpkg
+
+import (
+	"compress/gzip"
+	"io"
+
+	"github.com/spf13/afero"
+)
+
+var _ io.ReadCloser = &gzipFileReader{}
+
+// gzipFileReader reads compressed contents from a file.
+type gzipFileReader struct {
+	f    afero.File
+	gzip *gzip.Reader
+}
+
+// GzipFileReader constructs a new gzipFileReader from the passed file.
+func GzipFileReader(f afero.File) (io.ReadCloser, error) {
```

Comment:
Does this need to take `afero.File` as an argument? At a glance it seems like an `io.ReadCloser` would work.

Reply:
Nope -- good catch! This can just be a `GzipReadCloser` 👍🏻